src/prefect/utilities/asyncutils.py

```diff
@@ -165,13 +177,12 @@ def run_async_in_new_loop(__fn: Callable[..., Awaitable[T]], *args: Any, **kwarg
     return anyio.run(partial(__fn, *args, **kwargs))
 
 
+def mark_as_worker_thread():
+    _thread_local.is_worker_thread = True
+
+
 def in_async_worker_thread() -> bool:
-    try:
-        anyio.from_thread.threadlocals.current_async_module
```

Comment:
this thing doesnt exist anymore in `anyio` - this is basically what we're taking on responsibility for doing now